balta/src/test/scala/za/co/absa/db/balta/classes/DBTableIntegrationTests.scala

```diff
@@ -0,0 +1,200 @@
+    table.insert(Params
+      .add("id_field", 2)
+      .add("text_field", "textB")
+      .add("boolean_field", false)
+    )
+    table.insert(Params
+      .add("id_field", 3)
+      .add("text_field", "textB")
+      .add("boolean_field", true)
+    )
+    table.insert(Params
+      .add("id_field", 4)
+      .add("text_field", "textA")
+      .add("boolean_field", false)
+    )
+    val insertedNulls = table.insert(Params
+      .add("id_field", 5)
+      .addNull("text_field")
+      .addNull("boolean_field")
+    )
+    insertedNulls.assertHasJustId(5)
+
+    assert(table.count() == 5)
+
+    val returnedValue = table.where("text_field != 'textA'", "id_field") { queryResult =>
+      queryResult.next().assertTo(2, "textB", booleanField = false)
+      queryResult.next().assertTo(3, "textB", booleanField = true)
+      assert(queryResult.noMore)
+      "Hello world"
+    }
+    assert(returnedValue == "Hello world")
+
+    table.where("text_field = 'textA' AND boolean_field") { queryResult =>
+      queryResult.next().assertTo(1, "textA", booleanField = true)
+      assert(queryResult.noMore)
+    }
+
+    val deletedRows = table.deleteWithCheck("text_field = 'textA'") { queryResult =>
+      queryResult
+        .toList
+        .sortBy(_.getLong("id_field").get)
+    }
+    assert(deletedRows.length == 2)
+    deletedRows.head.assertTo(1, "textA", booleanField = true)
+    deletedRows(1).assertTo(4, "textA", booleanField = false)
+
+    assert(table.countOnCondition("boolean_field") == 1)
+
+    val remainingRows = table.all() { queryResult =>
+      queryResult.toList.sortBy(_.getLong("id_field").get)
+    }
+    assert(remainingRows.length == 3)
+    remainingRows.head.assertTo(2, "textB", booleanField = false)
+    remainingRows(1).assertTo(3, "textB", booleanField = true)
+    remainingRows(2).assertHasJustId(5)
+
+    table.delete("text_field IS NOT NULL")
+    assert(table.count() == 1)
+    table.deleteWithCheck{ queryResult =>
+      queryResult.next().assertHasJustId(5)
+      queryResult.noMore
+    }
+    assert(table.count() == 0)
+  }
+
+}
+
+object DBTableIntegrationTests {
+  implicit class QueryResultRowAssertion(val row: QueryResultRow) extends AnyVal {
+    private def error(field: String, expected: Any): String = {
+      s"`$field` field does not contain [$expected] but [${row(field).orNull}]"
+    }
+
+    def assertTo(idField: Long, textField: String, booleanField: Boolean, createdAt: OffsetDateTime): Unit = {
+      assertTo(idField, textField, booleanField)
+      assert(row("created_at") @= createdAt, error("created_at", createdAt))
+    }
+
+    def assertTo(idField: Long, textField: String, booleanField: Boolean): Unit = {
+      assert(row("id_field") @=  idField, error("id_field", idField))
+      assert(row("text_field") @= textField, error("text_field", textField))
+      assert(row("boolean_field") @= booleanField, error("boolean_field", booleanField))
```

Comment:
Why @= instead of == in case of boolean?

Reply:
`row("field_name")` - and all the other getters - return an `Option`. Therefore _balta_ implemented an Option enhancement that introduces `@=` as a `contains` synonym.